usecase/implementation/src/main/java/eu/openg/guessnumberapi/usecase/implementation/GuessNumberInteractor.java

```diff
@@ -1,17 +1,34 @@
 package eu.openg.guessnumberapi.usecase.implementation;
 
-import eu.openg.guessnumberapi.gateway.api.GameEntityRepository;
+import eu.openg.guessnumberapi.domain.Game;
+import eu.openg.guessnumberapi.gateway.api.GameRepository;
 import eu.openg.guessnumberapi.usecase.api.BoundaryGuessResponse;
+import eu.openg.guessnumberapi.usecase.api.BoundaryGuessResultStatus;
 import eu.openg.guessnumberapi.usecase.api.GuessNumberUseCase;
 
+import static java.util.Objects.isNull;
+
 public class GuessNumberInteractor implements GuessNumberUseCase {
 
-    GuessNumberInteractor(GuessValidator gateway, GameEntityRepository gameEntityRepository) {
+    private final GuessValidator gateway;
+    private final GameRepository gameRepository;
+
+    public GuessNumberInteractor(GuessValidator gateway, GameRepository gameRepository) {
+        this.gateway = gateway;
+        this.gameRepository = gameRepository;
     }
 
     @Override
     public BoundaryGuessResponse checkGuessAndReturnResponse(int gameId, int guessNumber) {
-        //TODO complete
-        throw new UnsupportedOperationException();
+        Game game = gameRepository.fetchGameEntity(gameId);
```

Comment:
čia irgi galima su optional, kaip route: return Optional.of(gameRepository).map....